inspirehep/modules/literaturesuggest/tasks.py

```diff
@@ -24,205 +24,220 @@
 
 import copy
 
-from datetime import date
-
-from flask import current_app
 
 from sqlalchemy.orm.exc import NoResultFound
 
-from idutils import is_arxiv_post_2007
 from invenio_accounts.models import User
 from invenio_oauthclient.models import UserIdentity
+
+from idutils import is_arxiv_post_2007
 from inspirehep.modules.forms.utils import filter_empty_elements
-from inspirehep.modules.search.api import JournalsSearch
 from inspirehep.utils.record import get_title, get_value
 
-from .dojson.model import literature
+from inspire_schemas.api import LiteratureBuilder
+
+from inspirehep.modules.search.api import JournalsSearch
+from inspirehep.utils.pubnote import split_page_artid
 
 
 def formdata_to_model(obj, formdata):
     """Manipulate form data to match literature data model."""
+    def _language_mapping(language, other_language):
+        languages = [('en', 'en'),
+                     ('rus', 'ru'),
+                     ('ger', 'de'),
+                     ('fre', 'fr'),
+                     ('ita', 'it'),
+                     ('spa', 'es'),
+                     ('chi', 'zh'),
+                     ('por', 'pt'),
+                     ('oth', 'ot')]
+
+        if language == 'oth' and other_language:
+            return other_language
+
+        return unicode(dict(languages).get(language))
```

Comment:
Bigger problem: the form accepts a free-form language, while the schema wants ISO 639-1 two-letter codes. Right now (tested on QA), `other_language` gets put into the record. This will not validate with the newer schema. Maybe we should use something like [pycountry](https://pypi.python.org/pypi/pycountry) to extract the language? Ideally the form should autocomplete languages and transform them to ISO 639-1. What do you think @jmartinm ?